File: ParkingObserverC#/ParkingObserver/Observer/Estacionamento.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ParkingObserver.Observer
{
   public abstract class Estacionamento //class Stock
    {
        public string Symbol { get; set; }
        private int _numVagas;

        private List<ICarros> _carros = new List<ICarros>();

        // Constructor
        public Estacionamento(string symbol, int numVagas)
        {
            this.Symbol = symbol;
            this._numVagas = numVagas;
        }

        public void Attach(ICarros carro)
        {
            _carros.Add(carro);
        }

        public void Detach(ICarros carro)
        {
            _carros.Remove(carro);
        }

        public void MudarDeAndar(string nomeCarro, int novoAndar)
        {
            foreach (Carros car in _carros)
            {
                if (car.Nome == nomeCarro)
                {
                    car.Andar = novoAndar;
                }

            }
        }
        public void Notify()
        {
            foreach (ICarros carro in _carros)
            {
                carro.Update(this);
            }
            Console.WriteLine("");
        }

        // Gets or sets the price

        public int NumeroDeVagas
        {
            get { return _numVagas; }

            set
            {
                if (_numVagas != value)
                {
                    _numVagas = value;
                    Notify();
                }
            }
        }
    }
}

File: ParkingObserverJava/src/br/Observer/Estacionamento.java
package br.Observer;

import java.util.ArrayList;
import java.util.List;

public abstract class Estacionamento {

	public String Symbol;
	private int NumVagas;
	
	private List<Carros>_carros = new ArrayList<Carros>();
	
	public String getSymbol() {
		return Symbol;
	}
	public void setSymbol(String symbol) {
		Symbol = symbol;
	}
	
	public Estacionamento(String symbol, int numVagas)
	{
	this.Symbol= symbol;
	this.NumVagas = numVagas;
	}
	
	public int getNumVagas() {
		return NumVagas;
	}
	public void setNumVagas(int numVagas) {	
		if(NumVagas != numVagas)
		{
		NumVagas = numVagas;
		Notify();
		}
	}
	public void Attach(Carros carro)
	{
		_carros.add(carro);
	}
	public void Detach(Carros carro)
	{
		_carros.remove(carro);
	}
	public void MudarDeAndar(String nomeCarro, int novoAndar)
	{
		for (Carros car : _carros) {
			if(car.Nome == nomeCarro)
			{
				car.Andar = novoAndar;
			}
		}
	}

	public void Notify()
	{
	for (ICarros carro : _carros) {
		carro.Update(this);
	}
	
	}
}
